src/components/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,31 @@
+import { ToggleButtonGroup, ToggleButton as ToggleButtonComponent } from '@mui/material/';
+import { ViewModule, ViewList } from '@mui/icons-material';
+import { DataVisual, ToggleButtonT } from '../../types/types';
+
+const ToggleButton = ({ dataVisual, setDataVisual }: ToggleButtonT) => {
+  const handleChange = (
+    _event: React.MouseEvent<HTMLElement, MouseEvent>,
+    dataVisual: DataVisual
+  ) => {
+    if (dataVisual && dataVisual !== null) return setDataVisual(dataVisual);
```

Comment:
From the interface, `dataVisual` can only be `'grid' | 'list' | null` so I think that `if (dataVisual) {...}` is enough?